build.sbt

```diff
@@ -4,6 +4,16 @@ organization in ThisBuild := "me.shadaj"
 
 scalaVersion in ThisBuild := "2.12.8"
 
+Compile / scalaSource := baseDirectory.value / "src" /  "main" / "scala"
+
+Test / scalaSource := baseDirectory.value / "src" / "test" / "scala"
```

Comment:
Hmm weird that IntelliJ doesn't infer those Scala source locations on its own :thinking:

Reply:
Maybe because of the insane project in project structure. Anyway pointing them by hand solves the issue.

Reply:
Recent versions of IntelliJ are supposed to support Scala.js / Scala Native project structures. What's the type of errors you are facing?